.github/workflows/build_docker_image.yml

```diff
@@ -1,64 +1,81 @@
 name: Build Docker image
 
-on:
-  push:
-    branches: master
+on: [push, pull_request]
 
 jobs:
   main:
     name: Docker
-    if: github.repository_owner == 'XboxDev'
     runs-on: ubuntu-latest
     timeout-minutes: 20
     strategy:
       matrix:
-        image-name: [nxdk, nxdk-debug, nxdk-lto]
+        tag: [latest, debug, lto]
```

Comment:
I don't like these new names.

debug and lto are also latest, aren't they?

Reply:
https://github.com/XboxDev/nxdk/blob/a82c2c04bc92ab8e8f69eb1bd013452646dce96a/.github/workflows/build_docker_image.yml#L61-L67

@JayFoxRox, according to new changes, if latest tag is selected, then it will exclude suffix tag of the "tag" from matrix processing. Yet I agree with you for the nxdk tag replacement is confusing.

`latest` tag should be change to `release` tag to follow up the confusion JayFoxRox brought up for clarification. Since it is compiling a release build. Correct?

Reply:
So the tags and container names in the dockerhub and xboxdev group are not different with this PR compared to without this PR. Sorry if this looks a bit confusing.

The reason for these tags in the registry of the repo itself is that for docker containers the tag latest is the default tag, which docker will automatically use if no tag is provided. Alternative builds tend to have their own unversioned tag names or different container names (which is not possible here, since we can only use nxdk as container name in this case). The way I'm tagging here is in line with standard practises on the Docker Hub for example, so I think not using latest would cause more confusion.

Hopefully that clarifies why I picked these tag names. Let me know if you still want me to change it, then I can still do so.

Reply:
Hmm... According the doc https://docs.docker.com/engine/reference/commandline/tag/#tag-an-image-referenced-by-name, tag isn't what you think it is. It's more or less like a branch name or in this case image name. So, using matrix's tag as an image name isn't accurate reference.

Other flaw I see from this [PR changes is it push nxdk, the image, tag with (via latest, debug, lto)](https://github.com/sharkwouter/nxdk/pkgs/container/nxdk) as a commit tag. In order word, it messes with nxdk package as placeholder storage for the next matrix to able use it. And it will affect if multiple pushed/merged simultaneously which each will have different delay to start the process. I'm fairly sure this isn't the way we want it to do. Plus cause more confusion with nxdk's "nxdk" package's tags vs other packages which are currently in private I suspect. Using GH's artifact, cache, or other alternative are recommended to do this if we want to handle the docker image within GH's Action temporary before push to package registery as permanent storage.

With that said, I don't see the reason for need to remove GH actions? They're there to simplified the process. And unnecessary to do login every step. We can perform if check for each step to skip them.

Reply:
I see what you mean, I think I addressed this now unless I misunderstood. The latest commit removes the need to pull in between, resolving the race condition you mentioned.